plugins/http_plugin/include/eosio/http_plugin/common.hpp

```diff
@@ -164,8 +164,8 @@ inline auto make_http_response_handler(http_plugin_state& plugin_state, detail::
       plugin_state.bytes_in_flight += payload_size;
 
       // post back to an HTTP thread to allow the response handler to be called from any thread
-      boost::asio::post(plugin_state.thread_pool.get_executor(),
-                        [&plugin_state, session_ptr, code, payload_size, response = std::move(response), content_type]() {
+      boost::asio::dispatch(plugin_state.thread_pool.get_executor(),
```

Comment:
I'm having a hard time figuring out why `dispatch` is appropriate here? Whay can't we just execute the lambda here?

Reply:
`dispatch` is the one that allows execution of the lambda here. `post` is the one that make sure it always executes outside this thread.

Reply:
Do we really need to call `dispatch` instead of just executing the lambda's code here?

Reply:
This can be called from the application thread, in that case we want to do the json parsing on the http thread.